src/fuzzing.md

```diff
@@ -0,0 +1,132 @@
+# Fuzzing
+
+<!-- date-check: Mar 2023 -->
+
+For the purposes of this guide, *fuzzing* is any testing methodology that
+involves compiling a wide variety of programs in an attempt to uncover bugs in
+rustc. Fuzzing is often used to find internal compiler errors (ICEs). Fuzzing
+can be beneficial, because it can find bugs before users run into them and
+provide small, self-contained programs that make the bug easier to track down.
+However, some common mistakes can reduce the helpfulness of fuzzing and end up
+making contributors' lives harder. To maximize your positive impact on the Rust
+project, please read this guide before reporting fuzzer-generated bugs!
+
+## Guidelines
+
+### In a nutshell
+
+*Please do:*
```

Comment:
could also mention debug assertions here.

Reply:
I don't know about this - can you say more?

Reply:
When bootstrapping rust, you can enable debug assertions in your config.toml which then enables debug assertions in rustc https://github.com/rust-lang/rust/blob/master/config.example.toml#L438

(bugs that might still be useful to know, but sometimes are not that bad or discovering them automatically has a noticeable performance hit that we don't want to deal with all the time etc..)

https://github.com/rust-lang/rust/labels/requires-debug-assertions

Reply:
Great, I covered that a bit later on. Thanks for the info!